server/filestore.go

```diff
@@ -2274,7 +2273,7 @@ func (mb *msgBlock) firstMatchingMulti(sl *Sublist, start uint64, sm *StoreMsg)
 		}
 		expireOk := seq == lseq && mb.llseq == seq
 
-		if r := sl.MatchWithResult(fsm.subj, &result); len(r.psubs) > 0 {
+		if sl.HasInterest(sm.subj) {
```

Comment:
I think you still need to distinguish between plain subs and queue subs. The original code here is just checking for plain sub interest (`len (r.psubs) > 0`), which means that if there was only queue interest, the "if" statement would have evaluate to false. You need to check with @derekcollison if it was an omission or not. If not, then you need to have booleans/int to indicate which interest type you want to check.

Reply:
I think any interest should short circuit. For where this would be used heavily its only plain subs. We could use this for all permissions checks as well.